src/process/enumeration.rs

```diff
@@ -0,0 +1,85 @@
+use super::{Enumeration, EnumerationMember};
+use log::warn;
+
+use crate::api_map::APIMap;
+
+/// Convert a snake_case or SCREAMING_SNAKE_CASE string to PascalCase
+fn convert_casing(input: &String) -> String {
+    let mut result = String::new();
+    result.reserve(input.len()); // Pre-allocate memory to avoid re-allocations
+    let mut capitalize_next = true;
+
+    for c in input.chars() {
+        if c == '_' {
+            capitalize_next = true;
+        } else if capitalize_next {
+            result.push_str(&c.to_uppercase().to_string());
+            capitalize_next = false;
+        } else {
+            match c.to_lowercase().next() {
+                Some(c) => result.push(c),
+                None => result.push(c),
+            }
+        }
+    }
+
+    result
+}
+
+/// Remove the common string from the beginning of the identifier
+pub fn remove_common_string(input: &String, identifier: &String) -> String {
+    let mut input = input.to_lowercase();
+
+    if input.starts_with("_") {
+        input = input.replacen("_", "", 1);
+    }
+
+    let mut identifier = identifier.to_lowercase();
+
+    if identifier.starts_with("_") {
+        identifier = identifier.replacen("_", "", 1);
+    }
+
+    let mut input_iter = input.chars().peekable();
+    let mut identifier_iter = identifier.chars();
+
+    while input_iter.peek() == identifier_iter.next().as_ref() {
+        input_iter.next();
+    }
+
+    input_iter.collect()
+}
+
+pub fn enumeration_processor(api_map: &APIMap) -> Vec<Enumeration> {
+    let enumerations: Vec<Enumeration> = api_map
+        .enums
+        .clone()
+        .into_iter()
+        .map(|enumeration| Enumeration {
+            identifier: if enumeration.identifier.is_some() {
+                Some(convert_casing(&enumeration.identifier.clone().unwrap()))
+            } else {
+                warn!("Enumeration identifier is None");
```

Comment:
Idk if that log level is a good choice
A warning constitutes "this shit probably went wrong", but in this context it's more like a debug message, because anon enums are expected to exist
Oh, and yes, this part should probably use the new CLI args for anon enum processing

Reply:
Okay, but this configuration option is passed through the `Cli`, but only `Config::generation` is passed in the `make_hl_ast parameters. How do you want me to proceed?

Reply:
I made it in a way that easily allows just adding one more argument into the function, and later pass it down to any other function inside. Call it "dependency injection" of sorts 😅